api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -217,45 +220,54 @@ private CodeableConcept getManifestation(List<AllergyReaction> reactions) {
 			return null;
 		}
 		
-		if (globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_MILD, "")
-		        .equals(severityConcept.getUuid())) {
+		getSeverityConceptUuids();
+		
+		if (severityConceptUuids.isEmpty()) {
+			return null;
+		}
+		
+		if (severityConcept.getUuid().equals(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_MILD))) {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.MILD;
-		} else if (globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_MODERATE, "")
-		        .equals(severityConcept.getUuid())) {
+		} else if (severityConcept.getUuid().equals(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_MODERATE))) {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.MODERATE;
-		} else if (globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_SEVERE, "")
-		        .equals(severityConcept.getUuid())) {
+		} else if (severityConcept.getUuid().equals(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_SEVERE))) {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.SEVERE;
 		} else {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.NULL;
 		}
 	}
 	
 	private Concept getOpenmrsSeverity(AllergyIntolerance.AllergyIntoleranceSeverity severity) {
-		Concept concept;
+		if (severity == null) {
+			return null;
+		}
+		
+		getSeverityConceptUuids();
+		
+		if (severityConceptUuids.isEmpty()) {
+			return null;
+		}
+		
 		switch (severity) {
 			case MILD:
-				concept = conceptService
-				        .getConceptByUuid(globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_MILD))
+				return conceptService.getConceptByUuid(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_MILD))
 				        .orElse(null);
-				break;
 			case MODERATE:
-				concept = conceptService
-				        .getConceptByUuid(globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_MODERATE))
+				return conceptService.getConceptByUuid(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_MODERATE))
 				        .orElse(null);
-				break;
 			case SEVERE:
-				concept = conceptService
-				        .getConceptByUuid(globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_SEVERE))
+				return conceptService.getConceptByUuid(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_SEVERE))
 				        .orElse(null);
-				break;
 			case NULL:
 			default:
-				concept = conceptService
-				        .getConceptByUuid(globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_OTHER))
+				return conceptService.getConceptByUuid(severityConceptUuids.get(FhirConstants.GLOBAL_PROPERTY_OTHER))
 				        .orElse(null);
 		}
-		return concept;
 	}
 	
+	private void getSeverityConceptUuids() {
```

Comment:
Instead of setting a class variable, just return the map from this function and use it as a local variable where needed. Storing state in DAO objects is a bad idea (what happens if the DAO needs to be called simultaneously for two different queries?)